app/models/image_data.rb

```diff
@@ -47,4 +50,15 @@ def dimensions
                       Dimensions.new(image[:width], image[:height])
                     end
   end
+
+  def filename_is_unique
+    return if validate_on_edition.blank? || validate_on_image.blank? || file.blank?
+
+    edition = validate_on_edition
+    image = validate_on_image
+
+    if edition.images.excluding(image).any? { |i| i.filename == filename }
+      errors.add(:file, message: "called #{filename} has already been uploaded. Choose a different image or rename the file.")
```

Comment:
Is this filename normalised (i.e. has it had nasty characters removed and such)? Just wondering if this may come out with a different filename potentially to what the user uploaded.

On the otherhand, if it's not normalised I wonder if that's a problem as it could breaking embedding.

Reply:
We don't do anything specific to clean filenames – but CarrierWave does some default sanitization, so we implicitly use that.

https://github.com/carrierwaveuploader/carrierwave/blob/1531a67366f0e25e3d298133a72c81b6c9c0dc83/lib/carrierwave/sanitized_file.rb#L297-L305

Reply:
With the CarrierWave sanitization, there are instances where image_data will flag a duplicate filename error on filenames which seem unique to the user - for example `"my>file.jpg"` and `"my$file.jpg"` will both sanitize to `"my_file.jpg"`. To avoid confusing the user in this scenario, @rosaryou suggested an update to the upload guidance and error message.